src/network/core_utils.rs

```diff
@@ -34,6 +35,65 @@ impl CoreUtils {
         final_slice.join(":")
     }
 
+    pub fn get_macvlan_mode_from_string(mode: &str) -> u32 {
+        // Replace to constant from library once this gets merged.
+        // TODO: use actual constants after https://github.com/little-dude/netlink/pull/200
+        match mode {
+            "bridge" => constants::MACVLAN_MODE_BRIDGE,
+            "private" => constants::MACVLAN_MODE_PRIVATE,
+            "vepa" => constants::MACVLAN_MODE_VEPA,
+            "passthru" => constants::MACVLAN_MODE_PASSTHRU,
+            "source" => constants::MACVLAN_MODE_SOURCE,
+            // default to bridge
+            _ => constants::MACVLAN_MODE_BRIDGE,
```

Comment:
This should error. We should only default to bridge if the mode key does not exists in the network options.